src/main/java/org/opensearch/knn/common/KNNVectorUtil.java

```diff
@@ -42,4 +44,18 @@ public static boolean isZeroVector(float[] vector) {
         }
         return true;
     }
+
+    /**
+     * Creates an int overflow safe arraylist. If there is an overflow it will create a list with default initial size
+     * @param batchSize size to allocate
+     * @return an arrayList
+     */
+    public static <T> ArrayList<T> createArrayList(long batchSize) {
+        try {
+            return new ArrayList<>(Math.toIntExact(batchSize));
+        } catch (Exception exception) {
+            // No-op
```

Comment:
This seems risky. Shouldnt we just throw?

Reply:
The [intent](https://github.com/opensearch-project/k-NN/pull/1950#discussion_r1715603078). This is being defensive while being consistent with current approach of unchecked arraylist expansion. The risk is unchanged